### PR TITLE
Implement ResNet-50 encoder

### PR DESCRIPTION
**Pull request by mak-en**

### CODE REVIEW

File: src/models/prototype_model/encoders.py [hunk 1]
```diff
@@ -176,25 +176,162 @@ def preprocess(self, x: torch.Tensor) -> torch.Tensor:
...
+            0,
+            1,
+            0,
+            0,
+            1,
+            0,
+            0,
+            1,
+            0,
+        ]
```
Comment (ViacheslavDanilov) [inline src/models/prototype_model/encoders.py:334]: @mak-en 

150 lines of code to create 3 variables?

**Review by ViacheslavDanilov — CHANGES_REQUESTED** [review]
NACK

File: src/models/prototype_model/encoders.py (hunk 1 above)
Comment (mak-en) [reply src/models/prototype_model/encoders.py:334]: Autoformatting by Black.

Comment (mak-en) [reply src/models/prototype_model/encoders.py:334]: @ViacheslavDanilov

**Review by ViacheslavDanilov — APPROVED** [review]
LGTM

**Pull request merged**